gettsim/social_insurance.py

```diff
@@ -20,160 +21,148 @@ def soc_ins_contrib(df, tb):
     smoothly until the regular one is reached
 
     """
-
     # initiate dataframe, indices must be identical
 
     # a couple of definitions
 
     # As there is only one household, we selcet west_ost dependent paramter in the
     # beginning and place them in a seperate dictionary tb_ost.
-    if df["east"]:
-        westost = "o"
-    else:
-        westost = "w"
+    westost = "o" if df_row["east"].all() else "w"
     tb_ost = {}
     for val in ["bezgr_", "mini_grenze", "kvmaxek", "rvmaxek"]:
         tb_ost[val] = tb[val + westost]
 
-    # ssc["above_thresh_kv"] = df["m_wage"] > tb_ost["kvmaxek"]
+    # ssc["above_thresh_kv"] = inout["m_wage"] > tb_ost["kvmaxek"]
     #
-    # ssc["above_thresh_rv"] = df["m_wage"] > tb_ost["rvmaxek"]
+    # ssc["above_thresh_rv"] = inout["m_wage"] > tb_ost["rvmaxek"]
 
     # This is probably the point where Entgeltpunkte should be updated as well.
 
     # Check if wage is below the mini job grenze.
-    belowmini = df["m_wage"] < tb_ost["mini_grenze"]
+    belowmini = df_row["m_wage"] < tb_ost["mini_grenze"]
 
     # Check if wage is in Gleitzone / Midi-Jobs
-    in_gleitzone = tb["midi_grenze"] >= df["m_wage"] >= tb_ost["mini_grenze"]
+    in_gleitzone = (tb["midi_grenze"] >= df_row["m_wage"]) & (
+        df_row["m_wage"] >= tb_ost["mini_grenze"]
+    )
 
     # Calculate accordingly the ssc
-    if belowmini:
-        ssc = ssc_mini_job()
-    elif in_gleitzone:
-        ssc = tb["calc_midi_contrib"](df, tb)
+    if belowmini.all():
+        df_row.loc[:, OUT_COLS] = 0.0
+    elif in_gleitzone.all():
+        # TODO: Before and in 2003 tb["midi_grenze"] is 0 and therefore we won't reach
+        #  this.
+        df_row = tb["calc_midi_contrib"](df_row, tb)
     else:
-        ssc = ssc_regular_job(df, tb, tb_ost)
+        df_row = ssc_regular_job(df_row, tb, tb_ost)
 
     # Exception: since 2013, marginally employed people may pay pension
     # insurance contributions.
-    """
-    if yr > 2012:
-        ssc.loc[df["m_wage"].between(1, tb["mini_grenzew"]), "rvbeit"] = tb[
-            "grvbs_mini"
-        ] * np.maximum(175, df["m_wage"])
-    """
+    # """
+    # if yr > 2012:
+    #     ssc.loc[inout["m_wage"].between(1, tb["mini_grenzew"]), "rvbeit"] = tb[
+    #         "grvbs_mini"
+    #     ] * np.maximum(175, inout["m_wage"])
+    # """
     # Self-employed may insure via the public health and care insurance.
```

Comment:
Should we delete this bit altoghether? I added it following a discussion in the topics course, but it does not seem to be the default.

Reply:
Kicked this out